libraries/bezier-rs/src/utils.rs

```diff
@@ -179,6 +179,19 @@ pub fn do_rectangles_overlap(rectangle1: [DVec2; 2], rectangle2: [DVec2; 2]) ->
 	top_right1.x >= bottom_left2.x && top_right2.x >= bottom_left1.x && top_right2.y >= bottom_left1.y && top_right1.y >= bottom_left2.y
 }
 
+pub fn is_subpath_inside_polygon<PointId: crate::Identifier>(polygon: &[DVec2], subpath: &Subpath<PointId>) -> bool {
+	let polypath = Subpath::from_anchors_linear(polygon.to_vec(), true);
+	if !subpath.subpath_intersections(&polypath, Some(0.02), Some(0.05)).is_empty() {
+		return false;
+	}
+	for anchors in subpath.anchors() {
+		if !polypath.contains_point(anchors) {
+			return false;
+		}
+	}
```

Comment:
We should do benchmarks to compare, but I have a feeling this is faster and should be put first. I also suggest doing the cheap early-exit bounding box check that you had suggested, and coming up with other optimizations.

Please also put a comment describing the logical justification for each of these checks and how it narrows the possibility space to result in a correct answer.